fennel.lua

```diff
@@ -1323,14 +1323,21 @@ local function eval(str, options)
     return loader()
 end
 
+local function dofile_fennel(filename)
+    local f = assert(io.open(filename, "rb"))
+    local vals = tpack(eval(f:read("*all"), { filename = filename,
```

Comment:
Do the read; then close; then do the eval as a tail call. Should be much faster :)

Reply:
Good call; thanks.